pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -312,13 +314,14 @@ func GetProxyHeaders(node *model.Proxy, push *model.PushContext, class istionetw
 
 func GetProxyHeadersFromProxyConfig(pc *meshconfig.ProxyConfig, class istionetworking.ListenerClass) ProxyHeaders {
 	base := ProxyHeaders{
-		ServerName:                 EnvoyServerName,
-		ServerHeaderTransformation: hcm.HttpConnectionManager_OVERWRITE,
-		ForwardedClientCert:        hcm.HttpConnectionManager_APPEND_FORWARD,
-		IncludeRequestAttemptCount: true,
-		SuppressDebugHeaders:       false,
-		GenerateRequestID:          nil, // Envoy default is to enable them, so set nil
-		SkipIstioMXHeaders:         false,
+		ServerName:                  EnvoyServerName,
+		ServerHeaderTransformation:  hcm.HttpConnectionManager_OVERWRITE,
+		ForwardedClientCert:         hcm.HttpConnectionManager_APPEND_FORWARD,
+		IncludeRequestAttemptCount:  true,
+		IncludeResponseAttemptCount: true,
```

Comment:
I don't think we should be adding new default-on headers. The general consensus was that all headers should have been default _off_, but we keep them on for backwards compat. All new ones should be off by default though

Reply:
Yes, i am more confused by setting it on gateway.
why should we expose the inner details to external client

Reply:
> The general consensus was that all headers should have been default off, but we keep them on for backwards compat

I agree with the decision, but this one is a bit weird case.  Setting `IncludeRequestAttemptCount: true` already enables `x-envoy-attempt-count` for upstream. If we keep `IncludeResponseAttemptCount: false` by default, then the same header is on in one direction and off in the other direction.

Moreover, we'll have 2 choices with the `attemptCount.disable` property in the api. Either we re-enable `IncludeResponseAttemptCount` by default because  `attemptCount.disable` would be false by default. Or we'll need to create a new property in the api to handle different directions of the same header (for example attemptCount and attemptCountResponse). Which seems like we're making things complicated for the user.

I don't mind either of those approaches but this implementation felt like the most smooth path for this specific header.

Reply:
> Yes, i am more confused by setting it on gateway.
why should we expose the inner details to external client

@hzxuzhonghu I think the general consensus was that presence of a gateway doesn't exactly mean "outside of mesh". For example, it can still be mesh workloads after the gateway. Further, we're already exposing the same header in case of egress gateways (`IncludeRequestAttemptCount` is true by default).
So I tried to keep the behaviour consistent in both the directions, and then the user can choose to disable both the headers using the API implementation if they want to.

I can still remove this from the gateway if there's a strong reason for not doing it.

Reply:
You set IncludeForResponse, so i say external clients can get this header.

Reply:
It is not the only reason, @howardjohn 's comments above make sense to me

Reply:
Whatever we do it _must_ be disabled by default